New: Add hint to validate display attribute of summary tag (#2658)
webhintio/hint: packages/hint-validate-summary/README.md

# validate-summary (`@hint/hint-validate-summary`)

Summary tag should have display: list-item only.

## Why is this important?

> Explain why this hint is important for your users

Reviewer: We should fill out the rest of this documentation as well. Key things to capture are that changing `display` on a `<summary>` tag to anything other than `list-item` will cause the open/close icon to disappear in browsers which implement the latest version of the standard (e.g. Firefox).

Also good to include supporting links if possible (which I think were included in the original issue).

# validate-summary (`@hint/hint-validate-summary`)

> Summary tag should have display: list-item only.

Reviewer: ```suggestion
Checks if `display` applied to a `<summary>` tag will hide the open/close icon.
```

## Why is this important?

> Changing the display on a summary tag to anything other than list-item

Reviewer: ```suggestion
Changing the `display` on a `<summary>` tag to anything other than `list-item`
```

> will cause the open/closeicon to disappear in browsers which implement

Reviewer: ```suggestion
will cause the open/close icon to disappear in browsers which implement
```

> the latest version of the standard.(eg: Firefox).

Reviewer: ```suggestion
the latest version of the standard (e.g. Firefox).
```

Chrome has

> <a href="https://bugs.chromium.org/p/chromium/issues/detail?id=590014&q=component%3ABlink%3EHTML%3EDetails%20&colspec=ID%20Pri%20M%20Stars%20ReleaseBlock%20Component%20Status%20Owner%20Summary%20OS%20Modified">

Reviewer: Let's use a markdown link reference here:
```suggestion
[a bug open][chromium-summary-issue]
```

Then down in "Link Labels" near the end of the file you can add:
```md
[chromium-summary-issue]: https://bugs.chromium.org/p/chromium/issues/detail?id=590014
```

a bug open to track aligning with the spec and Firefox in the future.</a>

## What does the hint check?

> The hint examines any rule that target a `summary` element and change the

Reviewer: ```suggestion
The hint examines any CSS rule that targets a `<summary>` tag and changes the
```

> display to anything other than `list-item`.

Reviewer: ```suggestion
`display` to anything other than `display: list-item`.
```

### Examples that **trigger** the hint

> `Any summary tag that alters the display to anything other than list-item`

Reviewer: Drop the back-tick quotes and end with a `:` for both this and subsequent examples.
```suggestion
Any CSS rule targeting a `summary` tag that alters the display to anything other than `list-item`:
```

Example:

> ```summary {

Reviewer: Put the triple back-ticks on their own line and include `css` at the end of the opening set to trigger correct syntax highlighting where supported (for both this and subsequent examples). Similarly put the closing triple back-ticks on a new line as well.
```suggestion
```css
summary {
```

    display: block;
}```

`Class or id of summary tag that alters the display property`

Example:

```summary.foo {
    display: inline;
}```

### Examples that **pass** the hint

`If the summary tag does not change the display property at all`

Example:

```summary {
    margin: 10px;
}```

`If there is no summary tag in the css file`

## Can the hint be configured?

If this hint allows some configuration, please put the format and

> options for the user.

Reviewer: Also, we can just remove this section since we're not currently allowing this hint to be configured.
